src/observers/telemetryObserver.ts

```diff
@@ -16,6 +16,12 @@ export class TelemetryObserver {
             case EventType.UserSignInCompleted:
                 this.handleUserSignInCompleted(<UserSignInCompleted>event);
                 break;
+            case EventType.UserSignOutTriggered:
+                this.handleUserSignOutTriggered(<UserSignOutTriggered>event);
+                break;
+            case EventType.UserSignOutCompleted:
```

Comment:
Since there is no global config to route the event to which observer, and this `UserSignOutCompleted` has been handled in many observers, just wondering is there a way to send an event to a specific handler, not broadcasting? Should that be more efficient, not all handlers would go through all the events? This question is not related to this PR.

Reply:
For your question, the answer is no. that would be more efficient if we send the event to a specific handler(actually if we do, we even don't need to use the event-driven framework, we can directly call it as a function), but that would lose the advantage that the event sender does not need to care about the event consumer, which is the principle of event-driven.

Reply:
OK, I see your point.
But do we always need to follow this sender and consumer decoupling principle?
For instance within micro-services, the component needs to send messages to the other components instead of function call since they are not in the same application.
Is our extension that complex so we want to avoid function call?

Reply:
I mean, we do have function calls.
Then what is the principle to use the event-driven way, like invoking between UI/logging?telemetry and extension main component?

Reply:
I second the use of simple function calls. Just look at the amount of code needed to log a simple telemetry or update a simple piece of text.